static/code/svg.js

```diff
@@ -107,5 +117,144 @@ svgEditor.getString = function() {
   return source;
 };
 
+/**
+ * Update the toolbox visualization in response to the editor state.
+ * This function polls the editor state four times a second.
+ * @param {boolean=} force If true, force a redraw.
+ */
+svgEditor.updateToolbox = function(force) {
+  // Highlight the current mode button.
+  var mode = svgEditor.canvas.getMode();
+  if (force ||
+      (mode !== 'resize' && mode !== 'rotate' &&
+      mode !== svgEditor.updateToolbox.oldMode_)) {
```

Comment:
Why not just `svgEditor.mode_`?  (Or `.toolboxMode_`.)

Also: if `mode` hasn't changed, can't you bail out?  I see you are doing stuff below, but will it have any effect?